docs/tutorials/UAV_tutorial.py

```diff
@@ -0,0 +1,368 @@
+"""
+
+# %%
+# Overview
+# --------
+# Starting with GPS data from an instrumented UAV, we will generate range, bearing, and
+# elevation measurements (from a given radar position). We will use Stone Soup's simple
+# :class:`~.SingleTargetTracker` to perform the tracking. At this point we are primarly interested
+# in the necessary motion models that may be needed so the example is fairly simple, but
+# we want to be able to easily expand the simulation to handle more complex scenarios.
+#
+# Items to note:
+#
+# - Assumes a single target track, which simplifies track management
+# - There is no clutter, and no missed detections -> No Data Association
+# - Need an initiator and deleter for the
+# - GPS updates are 1 sec., we assumes radar revisit is the same (little unrealistic)
+#
+# We are assuming a ground based radar:
+#
+# - No need to use a platform or mount sensors on it.
+# - Radar has course elevation resolution and fine bearing resolution.
+# - Use range standard deviation of 3.14 m as a replacement for range resolution.
+
+
+
+# %%
+# Setup: transition model, measurement model, updater and predictor
+# -----------------------------------------------------------------
+
+import numpy as np
+from stonesoup.models.transition.linear import (
+    ConstantVelocity,
+    CombinedLinearGaussianTransitionModel
+    )
+from stonesoup.predictor.kalman import ExtendedKalmanPredictor
+from stonesoup.updater.kalman import ExtendedKalmanUpdater
+from stonesoup.models.measurement.nonlinear import (
+    CartesianToElevationBearingRange
+    )
+from stonesoup.types.array import CovarianceMatrix
+
+
+transition_model = CombinedLinearGaussianTransitionModel(
+    [ConstantVelocity(1.0),
+     ConstantVelocity(1.0),
+     ConstantVelocity(1.0)])
+
+# Model coords = elev, bearing, range. Angles in radians
+meas_covar = np.diag([np.radians(np.sqrt(10.0))**2,
+                      np.radians(0.6)**2,
+                      3.14**2])
+
+meas_covar_trk = CovarianceMatrix(1.0*meas_covar)
+meas_model = CartesianToElevationBearingRange(
+            ndim_state=6,
+            mapping=np.array([0, 2, 4]),
+            noise_covar=meas_covar_trk)
+predictor = ExtendedKalmanPredictor(transition_model)
+updater = ExtendedKalmanUpdater(measurement_model=meas_model)
+
+# %%
+# Define Function
+# ---------------
+# Reads in our pickle file, which contains a pandas dataframe
+import pandas as pd
+import pickle as pkl
+
+def readPkl(fdir, fname):
+    # Reads pkl data file and changes column headings as appropriate
+    data = pd.read_pickle(fdir+fname)
+    data.rename(columns={'raw_lon' : 'longitude',
+                         'raw_lat' : 'latitude',
+                         'alt' : 'altitude (m)'},
+                inplace = True)
+    data['Vx m/s'] = (np.sin(np.deg2rad(data['heading'])) *
+                      data['speed'])
+    data['Vy m/s'] = (np.cos(np.deg2rad(data['heading'])) *
+                      data['speed'])
+    return data
+
```

Comment:
I think rather than using pandas, you could just save this to CSV, and use the exsiting CSV reader:
```python
ground_truth_reader = CSVGroundTruthReader(
    path='UAV_Rot.csv',
    state_vector_fields=['longitude', 'Vx m/s', 'latitude', 'Vy m/s', 'altitude (m)'],
    time_field='time',
    path_id_field='groupNb',
)
```

Reply:
@sdhiscocks Sent you an email - I rebased off the master and I applied your change, but I get an error: AttributeError: 'GroundTruthState' object has no attribute 'measurement_model'

This occurs in the:
for time, tracks in tracker:

I also get the same error when I try to extract the groundtruth i.e.:
for ground_truth in ground_truth_reader.groundtruth_paths

Do have a tutorial on these generators? I'm having a hell of a time trying to debug them.

Reply:
Sounds like your not on current master. May need to do a `git pull` on master branch and rebase again

Reply:
I did a git pull from my branch, but it wasn't updating the remote tracked upstream/master. Anyways - got it working after getting the proper updates.

Reply:
@sdhiscocks When I run the new program and try to switch to a different branch, it complains that:
Unlink of file 'docs/tutorials/UAV_Rot.csv' failed. Should I try again? (y/n) n
warning: unable to unlink 'docs/tutorials/UAV_Rot.csv': Invalid argument

Is the file being closed properly? Or is there a proper way to close it?

Reply:
> @sdhiscocks When I run the new program and try to switch to a different branch, it complains that:
> Unlink of file 'docs/tutorials/UAV_Rot.csv' failed. Should I try again? (y/n) n
> warning: unable to unlink 'docs/tutorials/UAV_Rot.csv': Invalid argument
> 
> Is the file being closed properly? Or is there a proper way to close it?

So the [code should be using context manager](https://github.com/dstl/Stone-Soup/blob/8ab9fce3411e4e2e4ca49f88560ff2bfb250e899/stonesoup/reader/generic.py#L77), but maybe that only works if the whole file is yielded??

Certainly should close the file if you stop the process/kill the kernel.

Reply:
I updated the tutorial based on Steven's changes to use Stone Soup components. This removes the pandas dependency.